rust-runtime/smithy-json/src/deserialize.rs

```diff
@@ -285,20 +211,28 @@ impl<'a> JsonTokenIterator<'a> {
 
     /// Expects a literal `null` next in the stream.
     fn expect_null(&mut self) -> Result<Token<'a>, Error> {
+        let offset = self.offset();
         self.expect_literal(b"null")?;
-        Ok(Token::ValueNull)
+        Ok(Token::ValueNull { offset })
     }
 
     /// Expects a boolean `true` / `false` to be next in the stream and returns its value.
     fn expect_bool(&mut self) -> Result<Token<'a>, Error> {
+        let offset = self.offset();
         match self.peek_expect()? {
             b't' => {
                 self.expect_literal(b"true")?;
-                Ok(Token::ValueBool(true))
+                Ok(Token::ValueBool {
+                    offset,
+                    value: true,
+                })
             }
             b'f' => {
                 self.expect_literal(b"false")?;
-                Ok(Token::ValueBool(false))
+                Ok(Token::ValueBool {
+                    offset,
+                    value: false,
+                })
             }
             _ => unreachable!(),
```

Comment:
probably worth commenting why this is unreachable or putting a message in `unreachable`
```suggestion
            _ => unreachable!("this function MUST only be called when the next byte is t or f"),
```